build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -54,300 +54,306 @@ let wrapCounter = 0;
 let transform;
 
 /**
- * Updates the set of preprocessors to run on HTML and JS files before testing.
- * Notes:
- * - The HTML transform is lazy-required because the server is built at startup.
- * - We must use babel on windows until esbuild can natively downconvert to ES5.
- * @param {!RuntimeTestConfig} config
+ * @extends {karmaConfig}
  */
-function updatePreprocessors(config) {
-  const createHtmlTransformer = function () {
-    return function (content, _file, done) {
-      if (!transform) {
-        const outputDir = `../../../${SERVER_TRANSFORM_PATH}/dist/transform`;
-        transform = require(outputDir).transformSync;
-      }
-      done(transform(content));
-    };
-  };
-  createHtmlTransformer.$inject = [];
-  config.plugins.push({
-    'preprocessor:htmlTransformer': ['factory', createHtmlTransformer],
-  });
-  config.preprocessors[karmaHtmlFixturesPath] = ['htmlTransformer', 'html2js'];
-  for (const karmaJsPath of karmaJsPaths) {
-    config.preprocessors[karmaJsPath] = ['esbuild'];
+class RuntimeTestConfig {
+  /**
+   * @param {string} testType
+   */
+  constructor(testType) {
+    this.testType = testType;
+    Object.assign(this, karmaConfig);
+    if (!this.plugins) {
+      this.plugins = [];
+    }
```

Comment:
Karma's `plugins` list is explicitly instantiated and can never be null, so this check is unnecessary.

https://github.com/ampproject/amphtml/blob/1dfbaffaba79483fa4a6fc7ead3ed1d4a77ee0a8/build-system/test-configs/karma.conf.js#L155-L171